update-dependency-pinning.sh

```diff
@@ -1,7 +1,9 @@
 #!/bin/bash
 
-WITNESS=app/witness.gradle
+# this will just append new checksums and not remove unnecessary ones
 
-echo "" > $WITNESS
-./gradlew -q calculateChecksums | grep -Ev "^(Skipping|Verifying)" | grep -Ev "files-2.1:|caches:transforms-3:|:build-tools:core-lambda-stubs.jar:|:platforms:android.jar:|-linux.jar:" > $WITNESS
+# to clean up the file, remove ./gradle/verification-metadata.xml,
+# run the command below and manually (re-)add checksums for missing operating systems windows, osx or linux for aapt2
+# checksums can be computed after downloading the respective jars of https://maven.google.com/web/index.html?q=aapt2#com.android.tools.build:aapt2
```

Comment:
Its a pity we can't do that automatically. Would be nice to have that file reflect all changes and not grow bigger over time.

Reply:
I agree, that would be better, but only adding new entries avoids removing manually added checksums as the ones for windows and osx for `aapt`. I think it will make sense anyway to look at the diff after running this command, so perhaps it is viable to remove old versions manually then. After all, this should be only updated after library upgrades.

Reply:
>  only adding new entries avoids removing manually added checksums as the ones for windows and osx for aapt.

but it won't help after we upgrade the gradle plugin as this will pull in new dependencies again that we would need to add manually, right?

> perhaps it is viable to remove old versions manually then

maybe easier to re-add windows/mac versions if we even want to support those?

Reply:
> > only adding new entries avoids removing manually added checksums as the ones for windows and osx for aapt.
> 
> but it won't help after we upgrade the gradle plugin as this will pull in new dependencies again that we would need to add manually, right?

That's correct.


> > perhaps it is viable to remove old versions manually then
> 
> maybe easier to re-add windows/mac versions if we even want to support those?

We could certainly change the bash script to always remove the old verification data (as was done with the witness plugin before), but I would have to check if the order of the dependencies is deterministic to ensure sensible diffs.

Reply:
According to https://github.com/gradle/gradle/issues/11664 the entries are sorted, so diffs should be fine